src/bgg/circuit/eval.rs

```diff
@@ -29,13 +29,10 @@ impl<P: Poly> Evaluable for P {
         Self::from_coeffs(params, &coeffs)
     }
 
-    fn from_bits(params: &Self::Params, _: &Self, bits: &[bool]) -> Self {
-        let modulus = params.modulus();
-        let one_elem = <P::Elem as PolyElem>::one(&modulus);
-        let zero_elem = <P::Elem as PolyElem>::zero(&modulus);
-        let coeffs: Vec<P::Elem> = bits
+    fn from_digits(params: &Self::Params, _: &Self, digits: &[u32]) -> Self {
+        let coeffs: Vec<P::Elem> = digits
             .par_iter()
-            .map(|&bit| if bit { one_elem.clone() } else { zero_elem.clone() })
+            .map(|&digit| <P::Elem as PolyElem>::constant(&params.modulus(), digit as u64))
```

Comment:
if we  turn digit as u64 type, can we accept as `digits: &[u64]`?

Reply:
Yes, but I think `u32` is enough for digits because `digit` is less than `base`, which will be less than `2^32`.